charmcraft/commands/store/__init__.py

```diff
@@ -508,6 +513,8 @@ def _validate_template_is_handled(self, filepath):
     def run(self, parsed_args):
         """Run the command."""
         name = get_name_from_zip(parsed_args.filepath)
+        if hasattr(parsed_args, "name"):
+            name = parsed_args.name
```

Comment:
`parsed_args` will always have `name`, in `None` if the user didn't pass it. So you could just do `if parsed_args.name:`...